### PR TITLE
feat(ingest): automated term classification for snowflake

### PR DESCRIPTION
**Pull request by mayurinehate**
## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_schema.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 from datetime import datetime
 from typing import Dict, List, Optional
 
+import pandas as pd
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_schema.py:6]: nit: this should be a typing-only dep - see mypy's docs on `TYPE_CHECKING`

Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_schema.py:6]: also I believe this PR https://github.com/acryldata/datahub-classify/pull/6 is going to remove the pandas dep from datahub-classify

File: metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_v2.py [hunk 2]
```diff
@@ -1014,3 +1037,22 @@ def inspect_session_metadata(self, conn: SnowflakeConnection) -> None:
     # Stateful Ingestion Overrides.
     def get_platform_instance_id(self) -> str:
         return self.config.get_account()
+
+    # Ideally we do not want null values in sample data for a column.
+    # However that would require separate query per column and
+    # that would be expensive, hence not done.
+    def get_sample_values_for_table(self, conn, table_name, schema_name, db_name):
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_v2.py:1044]: does this repeat work that the profiler does?

File: metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_schema.py (hunk 1 above)
Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_schema.py:6]: We are using pandas to read from snowflake cursor and translate to pandas dataframe in snowflake- https://docs.snowflake.com/en/user-guide/python-connector-pandas.html#migrating-to-pandas-dataframes. so this is required dependency for snowflake connector. I am not sure how converting this to typing-only dep will be useful.

File: metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_v2.py (hunk 2 above)
Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_v2.py:1044]: Kind of, but not exactly. With current implementation(with GE), profile can get only limited number of sample values(upto 20).

File: metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_schema.py (hunk 1 above)
Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_schema.py:6]: got it - in this case it's fine then, but I don't want to have a hard dep on pandas when we add classification to other sources

**Review by hsheth2 — APPROVED** [review]

File: metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_schema.py (hunk 1 above)
Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_schema.py:6]: Makes sense. I believe that would need refractor in [classification_mixin](https://github.com/datahub-project/datahub/blob/master/metadata-ingestion/src/datahub/ingestion/glossary/classification_mixin.py#L105) as well. Maybe, we can replace `sample_data` input param with a callable function that takes column as input and returns list-like values structure as output ?

Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_schema.py:6]: Sure, although I suspect a dict {column name -> list[sample values]} is good enough for this

Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_schema.py:6]: seems fair.

**Pull request merged**